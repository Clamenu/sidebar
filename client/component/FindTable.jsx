import React from 'react';
import styles from '../sidebar.css';

const FindTable = () => (
  <div className={styles.buttonContainer}>
    <a className={styles.button}>Find a Table</a>
  </div>
);

export default FindTable;
