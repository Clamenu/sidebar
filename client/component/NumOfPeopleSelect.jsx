import React from 'react';
import styles from '../sidebar.css';

const NumOfPeopleSelect = () => (
  <span className={styles.peopleContainer}>
    <img src='images/people.PNG' className={styles.icon}/>
    <select defaultValue='two' className={styles.select}>
      <option value='one'>1 person</option>
      <option value='two'>2 people</option>
      <option value='three'>3 people</option>
      <option value='four'>4 people</option>
      <option value='five'>5 people</option>
    </select>
    <img src='images/downarrow.PNG' className={styles.arrowIcon}/>
  </span>
);

export default NumOfPeopleSelect;