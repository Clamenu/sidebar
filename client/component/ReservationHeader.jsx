import React from 'react';
import styles from '../sidebar.css';

const ReservationHeader = () => (
  <div className={styles.headerContainer}>
    <h3 className={styles.header}>
      <img src='images/header.PNG'/>
      <span className={styles.headerTitle} id='header-title'>Make a Reservation</span>
    </h3>
  </div>
);

export default ReservationHeader;