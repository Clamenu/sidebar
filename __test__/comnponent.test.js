import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ReservationHeader from '../client/component/ReservationHeader.jsx';
import FindTable from '../client/component/FindTable.jsx'
import NumOfPeopleSelect from '../client/component/NumOfPeopleSelect.jsx';
import App from '../client/App.jsx'

describe('React Component Testing', () => {

  describe('Reservation Header', () => {

    test('should contain text "Make a Reservation"', () => {
      const component = shallow(<ReservationHeader />);
      const text = component.find('span').text();
      expect(text).toBe('Make a Reservation');
    });

  });

  describe('Find a Table Button', () => {

    test('should contain value "Find a Table"', () => {
      const component = shallow(<FindTable />);
      const value = component.find('input[type="submit"]').props().value;
      expect(value).toBe('Find a Table');
    });

  });

  describe('Number of People select', () => {

    test('should match previous snapshots', () => {
      const component = shallow(<NumOfPeopleSelect />);
      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });

  // describe('Mount App', () => {

  //   test('should match previous snapshots', () => {
  //     const app = mount(<App />);
  //     expect(app).toBe('something')
  //   });

  // });

});