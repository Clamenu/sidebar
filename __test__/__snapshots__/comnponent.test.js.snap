// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component Testing Number of People select should match previous snapshots 1`] = `
<span
  className="people-container"
>
  <img
    src="images/people.PNG"
  />
  <select
    defaultValue="two"
  >
    <option
      value="one"
    >
      1 Person
    </option>
    <option
      value="two"
    >
      2 People
    </option>
    <option
      value="three"
    >
      3 People
    </option>
    <option
      value="four"
    >
      4 People
    </option>
    <option
      value="five"
    >
      5 People
    </option>
  </select>
  <img
    src="images/downarrow.PNG"
  />
</span>
`;
